### ISSUE TITLE
The flags field should always be the first or last field

### ISSUE THREAD

**Issue by eregon** [ruby/prism#1892]
Currently the flags field seems at random positions when looking at `config.yml`.
While the field position does not matter when reading from a node, it matters when creating a new node, as that's the same as the constructor arguments order (and kwargs wouldn't help, e.g. Java & C & JavaScript & Rust, etc don't have them).

The location or `startOffset, length` are OTOH always consistently at the end, which is great.
If the flags was always the last field, it could be just before the location or after.
I think just before the location would make sense to keep these "common" fields together at the end.

@kddnewton WDYT?

**Comment by kddnewton** [comment]
Yeah that's fine. Are you creating new nodes often? I haven't found that to be a common use case.

**Comment by eregon** [comment]
I'm creating new nodes for desugaring things like *Variable{And,Or,Operator}WriteNode in Java.
It avoids code duplication or the need to have many helper methods which would quickly get messy, for something that's fully safe to desugar.

**Comment by kddnewton** [comment]
I think we're going to end up exposing flags on every node eventually anyway since we'll have a common flag for static literal and one for newline. In this case I'd like to move them to the front so that they're just after the type.
